### PR TITLE
Add auto_restart on error

### PR DESCRIPTION
**Pull request by Tocutoeltuco**
This branch adds Client.auto_restart attribute, which will automatically restart the Client on any error but cancellation ones. I also slightly changed the behaviour of _run_event, dispatch and close methods. The first one now returns True if the event ran successfully, and if there was any cancellation error, it is propagated instead of silenced. The dispatch method now returns a task wrapping _run_event. It might return None if the event was not found. Also, the close method doesn't stop the event loop anymore.

I've added a restart method and split the start one into `connect` and `sendHandshake` new functions.

### CODE REVIEW

File: aiotfm/connection.py [hunk 1]
```diff
@@ -95,7 +95,16 @@ def __init__(self, name, client, loop=None):
...
 		except Exception as e:
-			self.client.dispatch('connection_error', self, e)
+			future = self.client.dispatch('connection_error', self, e)
+			if future is not None:
+				# This future is a wrapper for _run_event, which returns True if the event ran successfully, False otherwise.
+				# If it returns False, it already handled the auto_restart.
+				if not await future:
+					return
+			if self.client.auto_restart:
+				await self.client.restart()
```
Comment (Athesdrake) [inline aiotfm/connection.py]: The client should wait few seconds before restarting.

Comment (Athesdrake) [reply aiotfm/connection.py]: It should also trigger an event (on_restart I fit perfectly) to tell the developer the client is restarting.

File: aiotfm/client.py [hunk 2]
```diff
@@ -599,13 +608,46 @@ def dispatch(self, event, *args, **kwargs):
...
+		"""|coro|
+		Starts the client.
+
+		:param api_tfmid: :class:`int` or :class:`str` your Transformice id.
+		:param api_token: :class:`str` your token to access the API.
+		"""
+		if keys is not None:
+			self.keys = keys
+		else:
+			self.api_tfmid = api_tfmid
```
Comment (Athesdrake) [inline aiotfm/client.py]: I don't like storing the token as a Client attribute. But I don't see how it could be done otherwise.

**Comment by Blank3495** [comment]
STOP SPAMMING BAGUETTES ALL OVER MY NOTIFICATIONS

File: aiotfm/client.py (hunk 2 above)
Comment (Tocutoeltuco) [reply aiotfm/client.py]: The same thing happened with the user and password. I hate storing them here and that's why I disabled the run method + auto restart.

**Pull request merged**